suite/tests/linux/syscall_pwait.c

```diff
@@ -49,40 +49,73 @@
 #include <unistd.h>
 #include <time.h>
 #include <poll.h>
+#include <sched.h>
+#include "thread.h"
+#include "condvar.h"
 
 typedef struct {
     sigset_t *sigmask;
     size_t sizemask;
 } data_t;
 
+typedef struct {
+    pthread_t main_thread;
+    bool delay;
+} args_t;
+
 static struct timespec sleeptime;
+static void *ready_to_listen;
+static args_t a;
```

Comment:
I do not see any reason for this to be a global variable?

Reply:
I can't make it local in kick_off_child_signal because the child thread may be reading the pointer after its lifetime is over.